samples/seat_massager/provider/src/provider_impl.rs

```diff
@@ -0,0 +1,200 @@
+        request: Request<SubscribeRequest>,
+    ) -> Result<Response<SubscribeResponse>, Status> {
+        warn!("Got subscribe request: {request:?}");
+
+        Err(Status::unimplemented("subscribe has not been implemented"))
+    }
+
+    /// Unsubscribe implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Unsubscribe request.
+    async fn unsubscribe(
+        &self,
+        request: Request<UnsubscribeRequest>,
+    ) -> Result<Response<UnsubscribeResponse>, Status> {
+        warn!("Got an unsubscribe request: {request:?}");
+
+        Err(Status::unimplemented("unsubscribe has not been implemented"))
+    }
+
+    /// Get implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Get request.
+    async fn get(&self, request: Request<GetRequest>) -> Result<Response<GetResponse>, Status> {
+        let request_inner = request.into_inner();
+        let entity_id: String = request_inner.entity_id.clone();
+        let consumer_uri: String = request_inner.consumer_uri;
+
+        info!("Received a get request for entity id {entity_id}");
+
+        let properties: Arc<Mutex<ProviderProperties>> = self.properties.clone();
+
+        tokio::spawn(async move {
+            if entity_id == sdv::airbag_seat_massager::massage_airbags::ID {
+                let result = ProviderImpl::get_message_airbags(properties.clone());
+                if result.is_err() {
+                    warn!("Failed to get {} due to: {}", entity_id, result.unwrap_err());
+                    return;
+                }
+                let client_result = DigitalTwinConsumerClient::connect(consumer_uri).await;
+                if client_result.is_err() {
+                    warn!("Unable to connect due to {}", client_result.unwrap_err());
+                    return;
+                }
+                let mut client = client_result.unwrap();
+
+                let publish_request =
+                    tonic::Request::new(PublishRequest { entity_id, value: result.unwrap() });
+                let response = client.publish(publish_request).await;
+                if let Err(status) = response {
+                    warn!("Publish failed: {status:?}");
+                }
+            } else {
+                warn!("The entity id {entity_id} is not recognized.");
+            }
+        });
+
+        let response = GetResponse {};
+
+        debug!("Completed the get request.");
+
+        Ok(Response::new(response))
+    }
+
+    /// Set implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Set request.
+    async fn set(&self, request: Request<SetRequest>) -> Result<Response<SetResponse>, Status> {
+        let request_inner = request.into_inner();
+        let entity_id: String = request_inner.entity_id.clone();
+        let value: String = request_inner.value;
+
+        info!("Received a set request for entity id {entity_id}");
+
+        let properties: Arc<Mutex<ProviderProperties>> = self.properties.clone();
```

Comment:
the type annotation here is not necessary

Reply:
I prefer to make it explicit.  Leaving it as is.